spec/hydra/works/services/virus_checker_service_spec.rb

```diff
@@ -38,17 +41,47 @@
     end
   end
 
-  context 'with a clean file' do
+  context 'with a clean unsaved file' do
     context 'that responds to :path' do
       it 'will return true' do
         expect(system_virus_scanner).to receive(:infected?).with('/tmp/file.pdf').and_return(false)
         allow(file).to receive(:path).and_return('/tmp/file.pdf')
         expect(virus_checker.file_has_virus?).to eq(false)
       end
     end
+
+    context "that does not respond to path" do
+      before do
+        allow(file).to receive(:respond_to?).with(:path).and_return(false)
+      end
+      context 'that the content responds to path' do
+        let(:content) { instance_double "File", read: ["abc123", nil], rewind: true, path: "abc123" }
+
+        it 'will return true' do
+          expect(system_virus_scanner).to receive(:infected?).with(kind_of(String)).and_return(false)
+          expect(content).not_to receive(:read)
+          expect(virus_checker.file_has_virus?).to eq(false)
+        end
+      end
+      context "that the content does not respond to path" do
+        let(:content) { instance_double "File", read: ["abc123", nil], rewind: true }
+        it 'will return true' do
+          expect(system_virus_scanner).to receive(:infected?).with(kind_of(String)).and_return(false)
+          expect(file).not_to receive(:stream)
+          expect(virus_checker.file_has_virus?).to eq(false)
```

Comment:
The "will return true" is in contradiction of the `expect(virus_checker.file_has_virus?).to eq(false)`